helpick-service/src/main/java/com/infoshare/service/NeedRequestService.java

```diff
@@ -26,27 +34,82 @@ public NeedRequestService(DB db) {
     public void createNeedRequest(String name, String location, String phone, TypeOfHelp typeOfHelp) {
         PersonInNeed personInNeed = new PersonInNeed(name, location, phone);
         NeedRequest needRequest = NeedRequest
-            .create(typeOfHelp, personInNeed);
+                .create(typeOfHelp, personInNeed);
         db.saveNeedRequest(needRequest);
     }
 
+    public void changeRequestStatus(List<NeedRequest> filteredList, int choice) {
+        try {
+            List<NeedRequest> activeNeedRequests = db.getNeedRequests();
+            NeedRequest changedRequest = filteredList.get(choice-1);
+            changedRequest.setHelpStatus(HelpStatuses.INPROGRESS);
+            changedRequest.setStatusChange(new Date());
+            for (int i = 0; i < activeNeedRequests.size(); i++) {
+                if (activeNeedRequests.get(i).getUuid().equals(changedRequest.getUuid())) {
+                    activeNeedRequests.set(i, changedRequest);
+                }
+            }
+            FileWriter writer = new FileWriter("NeedRequest.csv", false);
+            for (NeedRequest nr : activeNeedRequests) {
+                db.saveNeedRequest(nr);
+            }
+            writer.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void restoreStatusForExpiredRequests() {
```

Comment:
Można skorzystać ze stream i uprościć metodę:
```suggestion
     private boolean isNeedRequestOlderThan24HoursAndIsInProgress(NeedRequest needRequest) {
        return needRequest.getHelpStatus().equals(HelpStatuses.INPROGRESS) &&
                Duration.between(needRequest.getStatusChange().toInstant(), Instant.now()).toMinutes() > 1440;
    }

    public void updateRequestsStatus() {
        db.getNeedRequests().stream()
                .filter(this::isNeedRequestOlderThan24HoursAndIsInProgress)
                .peek(needRequest -> log.info("Found difference > 24h in request " + needRequest.getUuid() + ", changing status..."))
                .forEach(needRequest -> updateNeedRequestStatusAndDateAndSave(needRequest, HelpStatuses.NEW));
    }
```

Reply:
Dzięki, ale zostawiam swoją wersję.